### PR TITLE
Introduction overview editorial changes

### PR DESCRIPTION
**Pull request by glyn**

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -371,45 +373,45 @@ A JSONPath expression may use a combination of bracket and dot notations.
...
-This document treats the bracket notations as canonical and defines various shorthands in terms
+This document treats bracket notation as canonical and defines various shorthands in terms
 of bracket notation. Examples and descriptions use shorthands where convenient.
 
 ### Selectors
 
-A wildcard `*` ({{wildcard}}) in the expression `[*]` selects all children of an
-object or an array and in the expression `..[*]` selects all descendants of an object or an array.
+A wildcard `*` ({{wildcard}}) in the expression `[*]` selects all children of a
+value and in the expression `..[*]` selects all descendants of a value.
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:382]: Can we use "node" here?
"value" is seriously confusing.

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -371,45 +373,45 @@ A JSONPath expression may use a combination of bracket and dot notations.
...
 | `@`                 | [current node identifier](#filter-selector) (valid only within filter selectors)                                          |
-| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of JSON objects and arrays; contains one or more selectors, separated by commas        |
+| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of a value; contains one or more selectors, separated by commas        |
 | `.name`             | shorthand for `['name']`                                                                                                |
 | `.*`                | shorthand for `[*]`                                                                                                     |
-| `..[<selectors>]`   | [descendant segment](#descendant-segment): selects zero or more descendants of JSON objects and arrays; contains one or more selectors, separated by commas |
+| `..[<selectors>]`   | [descendant segment](#descendant-segment): selects zero or more descendants of a value; contains one or more selectors, separated by commas |
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:406]: dito

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -371,45 +373,45 @@ A JSONPath expression may use a combination of bracket and dot notations.
...
 
-{{tbl-overview}} provides a quick overview of the JSONPath syntax elements.
+{{tbl-overview}} provides a brief overview of JSONPath syntax.
 
 | Syntax Element      | Description                                                                                                             |
 |---------------------|-------------------------------------------------------------------------------------------------------------------------|
 | `$`                 | [root node identifier](#root-identifier)                                                                                |
 | `@`                 | [current node identifier](#filter-selector) (valid only within filter selectors)                                          |
-| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of JSON objects and arrays; contains one or more selectors, separated by commas        |
+| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of a value; contains one or more selectors, separated by commas        |
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:403]: dito

**Review by cabo — COMMENTED** [review]
I think that "a value" is really confusing to readers who haven't debated about our terminology for at least a year.  Can we use "node"?

File: draft-ietf-jsonpath-base.md (hunk 1 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:382]: Sure.

File: draft-ietf-jsonpath-base.md (hunk 2 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: Done.

File: draft-ietf-jsonpath-base.md (hunk 3 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: Done.

File: draft-ietf-jsonpath-base.md (hunk 1 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: Done.

**Comment by glyn** [comment]
@cabo: Thanks for those suggestions. I rebased on HEAD and applied them.

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -382,41 +384,41 @@ of bracket notation. Examples and descriptions use shorthands where convenient.
...
+node and in the expression `..[*]` selects all descendants of a node.
 
 An array slice `start:end:step` ({{slice}}) selects a series of
 elements from an array, giving a start position, an end position, and
-possibly a step value that moves the position from the start to the
+an optional step value that moves the position from the start to the
 end.
 
-Filter expressions `?<boolean expr>` select certain children of an object or array as in
+Filter expressions `?<boolean expr>` select certain children of a value, as in:
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:395]: The old text said that filter expressions are useful for containers; I think that information should stay there (and, if not, we should say node, not value).

File: draft-ietf-jsonpath-base.md [hunk 5]
```diff
@@ -382,41 +384,41 @@ of bracket notation. Examples and descriptions use shorthands where convenient.
...
 
-{{tbl-overview}} provides a quick overview of the JSONPath syntax elements.
+{{tbl-overview}} provides a brief overview of JSONPath syntax.
 
 | Syntax Element      | Description                                                                                                             |
 |---------------------|-------------------------------------------------------------------------------------------------------------------------|
 | `$`                 | [root node identifier](#root-identifier)                                                                                |
 | `@`                 | [current node identifier](#filter-selector) (valid only within filter selectors)                                          |
-| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of JSON objects and arrays; contains one or more selectors, separated by commas        |
+| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of a mode; contains one or more selectors, separated by commas        |
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:409]:
```suggestion
| `[<selectors>]`     | [child segment](#child-segment) selects zero or more children of a node; contains one or more selectors, separated by commas        |
```

File: draft-ietf-jsonpath-base.md [hunk 6]
```diff
@@ -382,41 +384,41 @@ of bracket notation. Examples and descriptions use shorthands where convenient.
...
-| `..[<selectors>]`   | [descendant segment](#descendant-segment): selects zero or more descendants of JSON objects and arrays; contains one or more selectors, separated by commas |
+| `..[<selectors>]`   | [descendant segment](#descendant-segment): selects zero or more descendants of a node; contains one or more selectors, separated by commas |
 | `..name`            | shorthand for `..['name']`                                                                                              |
 | `..*`               | shorthand for `..[*]`                                                                                                   |
 | `'name'`            | [name selector](#name-selector): selects a named child of an object                                                     |
-| `*`                 | [wildcard selector](#name-selector): selects all children of an array or object                                         |
+| `*`                 | [wildcard selector](#name-selector): selects all children of a value                                                    |
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:416]: value ➔ node (see above)

Comment (cabo) [reply draft-ietf-jsonpath-base.md:416]:
```suggestion
| `*`                 | [wildcard selector](#name-selector): selects all children of a node                                                    |
```

**Review by cabo — APPROVED** [review]

File: draft-ietf-jsonpath-base.md (hunk 4 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:395]: Good point. Fixed.

**Pull request merged**